CONTRIBUTING.md

```diff
@@ -26,12 +26,45 @@ Please try to spend some time on a good title and description of the issue.
 
 ## Pull requests
 
-If it is not a fork branch, please use the following format for your branch: `{gatsby,drupal}/{feature,bugfix,content}/slug`. 
+If it is not a fork branch, please use the following format for your branch: `{gatsby,drupal}/{feature,bugfix,content}/slug`.
 
 All pull requests should undergo code review. The code review can be performed by any volunteer who has been active in the project for a long time.
 
 A Pull Request has to pass validation by Github Actions (Triggered Automatically)
 
+## Branching Workflow
+
+### Branch overview
+
+* `develop`, main integration and testing branch
+* `master`, release branch that gets deployed to production
+* feature branches
+
+### Basic rules
+
+* All work starts in a feature branch
+* Feature branches are merged into `develop` for integration and testing
+* Once `develop` is stabilized and tested, it’s merged into `master`
+* All production deployments are done from `master`
+
+### Tips
+
+* If you need an urgent fix in production, create a hotfix branch on `master` (and then backport the fix into `develop`).
+* Never commit directly to `master` or `develop`, merges only.
+* Never merge `develop` to feature branches.
```

Comment:
Why is that? 🤔 I think sometimes this can be a viable case... am I missing smthing?

Reply:
If you merge the integration branch into the feature branch, you may bring in unrelated changes, which means the feature branch then no longer represents a single coherent topic. I have found it best to base the feature branches on the release branch (the last release), then you can easily mix & match them into the next release as needed. But these are all mostly suggestions – it’s OK to break the rules if you know what you’re doing.

Reply:
@zoul In that case you should change:

All work starts in a feature branch

to something like:
Start by creating feature branch from `master`, not from `develop`. This is to ensure feature branch represents a single coherent topic and release Engineer can easily mix & match them into the next release as needed.

Because from reading this CONTRIBUTING I'd start by creating fork of `develop` branch.

Reply:
But this way the integration will be a pain. Imagine me doing feature A -> merge to develop. Then someone comes, wants to do a feature B, so he starts from master once again. Now he realized he needs changes from feature A. So the solution is to merge the feature branch A into his feature branch B? 🤷‍♂️ I am not used to such flow, but mby that could work.. But id definitely personally rather put all of the changes to develop to make sure everything is integrated together. Then that branch is deployed to test, tested properly and then eventually prod...

Reply:
@rostaklein Thats heavily philosophical problem (effectively unrelated to this discussion). What what you're saying would effectively lead to `develop` becoming a `mega-feature-branch`.

Specifically addressing your `feature A` vs `feature B` example, when `feature A` is merged to develop, why do you (as a developer of `feature B`) start working on `feature B` when code you depends on (in `feature A`) is not merged into production yet and might never be merged there => which would render your work on `feature B` useless. Work on `feature B` in this case should not start before `feature A` is finalized, approves and merged to production.

Zoul's solution would also make it possible to pick features already merged to `develop` that gets deployed => or effectively drop feature at that point and never merge it into production (this is mentioned by Tomas on the branching Slack Thread as requirement). Where `mega-feature-branch` would not allow any changes in history (it'd be like blockchain!)

Reply:
Rosťa, the changes are eventually all merged into the development branch and tested together before releasing. But yes, you’re right, it’s sometimes more work, like when you want to check your yet-unmerged work together with some other changes. (It’s fairly easy to mix and test whatever you want in a private temporary integration branch, but it’s work.) And dependencies between feature branches can be a pain, yes.

I suggest to dodge the issue now by only _suggesting_ the “base feature branches on master” workflow. We can see how that works and tweak the process as needed.